lgc/builder/BuilderImpl.cpp

```diff
@@ -399,3 +347,66 @@ Value *BuilderImplBase::scalarize(Value *value0, Value *value1, Value *value2,
   Value *result = callback(value0, value1, value2);
   return result;
 }
+
+// =====================================================================================================================
+// Track back through to find the descriptor load to provide the indices for the waterfall or readfirstlane
+//
+// @param descUserInst : The instruction to get operands for descriptors
+// @param operandIdxs : The operand index/indices for the descriptor(s) to be checked
+// @param foundIndices : The index/indices used in waterfall loop or pass through readfirstlane
+void BuilderImplBase::findDescriptorIndices(Instruction *descUserInst, ArrayRef<unsigned> operandIdxs,
+                                            SmallVector<Value *, 2> &foundIndices) {
```

Comment:
This should use a `SmallVectorImpl<Value *>` instead, so that the callee doesn't restrict the pre-allocation done by the caller.